cosmwasm/enclaves/execute/src/registration/onchain.rs

```diff
@@ -56,21 +120,83 @@ pub unsafe extern "C" fn ecall_authenticate_new_node(
         return NodeAuthResult::SignatureInvalid;
     }
 
-    let result = panic::catch_unwind(|| -> Result<Vec<u8>, NodeAuthResult> {
+    let mut target_public_key: [u8; 32] = [0u8; 32];
+
+    let (vec_cert, vec_quote, vec_coll) = split_combined_cert(cert, cert_len);
+
+    if vec_quote.is_empty() || vec_coll.is_empty() {
```

Comment:
This seems like a hacky way to tell whether we're validating for epid or dcap - can't we just add a field here to the function call with the type of attestation?

Reply:
I would like to refrain from changing the protocol (protobufs and etc.)
The argument "cert" is a single blob with 3 things serialized: epid certificate, dcap quote, and dcap collateral.